capsules/extra/src/ieee802154/driver.rs

```diff
@@ -522,12 +542,20 @@ impl framer::KeyProcedure for RadioDriver<'_> {
     /// level `level` and key ID `key_id`. If no such key matches, returns
     /// `None`.
     fn lookup_key(&self, level: SecurityLevel, key_id: KeyId) -> Option<[u8; 16]> {
-        self.keys.and_then(|keys| {
+        let res = self.keys.and_then(|keys| {
             keys[..self.num_keys.get()]
                 .iter()
                 .find(|key| key.level == level && key.key_id == key_id)
                 .map(|key| key.key)
-        })
+        });
+
+        if res.is_none() {
+            self.backup_key_procedure
+                .unwrap_or_panic()
+                .lookup_key(SecurityLevel::EncMic32, KeyId::Index(2))
```

Comment:
I _think_ it'd be "rusty-er" to put the body of this `if` as closure of a chained `.unwrap_or_else(|` and omit the `res` variable.